test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -1190,57 +1191,55 @@ public async Task TableNamedRecord()
             }
         }
 
-// TODO: Port this test to .NET Core somehow
-#if NET461
         [Test, IssueLink("https://github.com/npgsql/npgsql/issues/392")]
         public async Task NonUTF8Encoding()
         {
-            using (var adminConn = await OpenConnectionAsync())
+#if !NET461
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+#endif
```

Comment:
We no longer target .NET Framework - so this isn't needed, no?

```suggestion
            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Reply:
BTW I wonder if we should (or even can) unregister the provider when the test completes (and mark it non-parallelizable), as it may affect global state etc... Probably not needed - just thinking out loud.

Reply:
The global state that is changed is which encodings you can actually get via `Encoding.GetEncoding()`.
The default in .NET core is only ASCII and the Unicode encodings. What we do here is registering the provider for some additional ones.
The only problem that could result from this is that we use some non-default encoding in Npgsql and fail to recognize that it isn't available by default if the respective test would run after this test.
I think that this is pretty unlikely but your comment is still valid.